GOVERNANCE.md

```diff
@@ -131,3 +122,75 @@ the voting process should be considered.
+
+Note that neither of these roles have voting powers in conflict resolution, these roles are for the code base change approval process only.
+
+### Pull Request Flow
+
+The general flow for a pull request approval process is as follows:
+
+1. Author submits the pull request
+1. Reviewers and approvers for the applicable code areas review the pull request and provide feedback that the author integrates
+1. Reviewers and/or approvers signify their LGTM on the pull request
+1. An approver approves the pull request based on at least one LGTM from the previous step
+    1. Note that the approver can heavily lean on the reviewer for examining the pull request at a finely grained detailed level. The reviewers are trusted members and approvers can leverage their efforts to reduce their own review burden.
+1. An approver merges the pull request into the target branch (master, release, etc.)
+
+### Role Assignments
+
+#### Declarations
+
+All roles will be assigned by the usage of [`CODE-OWNERS`](CODE-OWNERS) files committed to the code base.
+These assignments will be initially be defined in a single file at the root of the repo and it will describe all assigned roles for the entire code base.
+As we incorporate automation (i.e. bots) into this change acceptance process in the future, we can reorganize this initial single owners file into separate files amongst the codebase as the automation necessitates.
+
+The format of the file can start with simply listing the reviewers and approvers for areas of the code base using a YAML format:
+
+```yaml
+areas:
+  feature-foo:
+    approvers:
+    - alice
+    - bob
+    reviewers:
+    - carol
+```
+
+#### Process
+
+All new reviewers and approvers must be nominated by someone (anyone) opening a pull request that adds the nominated person’s name to the appropriate [`CODE-OWNERS`](CODE-OWNERS) files in the appropriate roles.
+The maintainer team will approve this nomination in the standard voting and conflict resolution process.
+
+Note that new reviewer/approver nominations do not go through the standard pull request approval process described previously, only the maintainer team can approve additions of members to the change approval roles.
+
+### Permissions
+
+Role assignees will be made part of the following Rook organization teams with the given permissions:
+
+* **Reviewers:** added to Contributors team so they have write permissions to the repo to assign issues, add labels to issues, add issues to milestones and projects, etc. but cannot merge to protected branches such as `master` and `release-*`.
+* **Approvers:** added to the Contributors team and additionally added to the new Approvers team that will have access to merge to protected branches.
+
+### Automation
+
+This process can be further improved by automation and bots to automatically assign the PR to reviewers/approvers, add labels to the PR, and merge the PR.
```

Comment:
Until we have bots or some other mechanism in place to restrict access to certain storage providers, we expect the approvers have a high level of trust and due diligence before merging PRs. One storage provider could be unintentionally impacted by the merge of another storage provider PR.  One storage provider won't realize how the other storage provider is impacted unless it is caught by the integration tests. This is a topic for another day, but just wanted to put it out there as something  we need to solve to really grow the storage provider ecosystem. Even if storage providers have all the right intentions and due diligence in code reviews, we have to reduce that risk of impact across storage providers. More thoughts on this coming with the rook charter doc...

Reply:
Yes I agree that there will be trust involved in this policy until there is full automation support for enforcement.   The work per storage provider is largely isolated unless changes are being made to the common rook framework, which I have specifically called out in the CODE-OWNERS file.  So if changes are made there, individual storage providers do not have the authority to approve/merge on their own.

Of course, this isn't strictly enforced until we have automation, so trust will be required.

Reply:
FWIW, you could use Prow from the kubernetes project to enforce/automate this

Reply:
@jberkus The changes were really modeled after K8s, so the hope is to be able to use the same bots. Can you provide more insight to getting this setup, or who is the best contact? Thanks!

Reply:
Correct, the [Kubernetes code review process](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#code-review-using-owners-files) is linked in this document, showing exactly what this proposal was modeled after.